source/develop/release-management/features/virt/engine-dedicated-cpus.md

```diff
@@ -0,0 +1,102 @@
+---
+Title: Engine Dedicated CPUs
+Category: feature
+Authors: sabusale
+---
+
+# Engine Dedicated CPUs
+
+## Summary
+
+Once choosing dedicated CPU policy, The guest vCPUs will be strictly pinned to a set of host pCPUs
+(similarly to static CPU pinning). The set of pCPUs will be chosen to match the required guest CPU topology.
+
+Engine handles the pinning automatically when starting the VM as opposed to the manual pinning that is
+done by the user and is static.
+
+## Owner
+
+-   Name: Saif Abu Saleh (sabusale)
+-   Email: sabusale@redhat.com
+
+## Responsibilities
+
+VDSM:
+
+- Manages the available pinned CPUs
+
+Engine:
+- Schedule VM to run on best satisifing host
+- creates vCpu to pCpu mappings
+
+
+## Backend
+
+1. Add dedicated to CPU Pinning Policy enum
+
+2.  Scheduling
+    1. Start VM:
+        - Scheduling phase
+            - Score all hosts to find best matching host with enough pCPUs
+        - Run phase
+            - Create vCpu to pCpu mapping based on the host free pCPUs reported
+            - Update VM CurrentCpuPinning DB field with the mapping
+
+    2. Stop/Destroy VM:
+        - Clear VM dynamic CPU pinning field
```

Comment:
and the representation the engine should have per-host - we can't restore the pinning state on the host on run-vm, that would be too much of an overhead

Reply:
Done